### PR TITLE
Add more GoingToCamp recreation areas

### PR DESCRIPTION
**Pull request by acaloiaro**
This PR adds additional GTC recreation areas, partially fixes https://github.com/juftin/camply/issues/177

### CODE REVIEW

**Comment by juftin** [comment]
Nice! Would You update the Provider Docs: https://github.com/acaloiaro/camply/blob/add-more-gtc-rec-areas/docs/providers.md#goingtocamp

We're almost at the point where we have too many providers for GoingToCamp to list them out, what do you think?

**Comment by acaloiaro** [comment]
> Nice! Would You update the Provider Docs: https://github.com/acaloiaro/camply/blob/add-more-gtc-rec-areas/docs/providers.md#goingtocamp
> 
> We're almost at the point where we have too many providers for GoingToCamp to list them out, what do you think?

Yeah what do you think about reordering the IDs to be alphabetical? Do you think that ship has sailed and they're now part of the API, or do you think it's okay for rec area IDs to be unstable? 

My concern is that now the list is arbitrarily ordered.

File: camply/providers/going_to_camp/going_to_camp_provider.py [hunk 1]
```diff
@@ -35,44 +35,69 @@
...
+    "www.nlcamping.ca": RecreationArea(
+        recreation_area="Newfoundland & Labrador Provincial Parks",
+        recreation_area_id=14,
+        recreation_area_location="Newfoundland and Labrador, CA",
+    ),
+    "camping.bcparks.ca": RecreationArea(
+        recreation_area="BC Parks",
+        recreation_area_id=15,
+        recreation_area_location="British Columbia, CA",
+    ),
```
Comment (juftin) [inline camply/providers/going_to_camp/going_to_camp_provider.py:100]: Starting this in a Thread - for some reason GitHub makes it hard to have PR conversations on a thread.

I'm not worried about changing the IDs of GoingToCamp right now - we're out of season and it's a new provider. 

Previously when generating the documentation I just ran the command `camply recreation-areas --provider GoingToCamp` to get the RecArea - and then just pasted the response into the docs. That's why they're in a funky order (which is technically alphabetical).

For Fuzzy matching, I'm a fan of [RapidFuzz](https://github.com/maxbachmann/RapidFuzz) - although it's written in C++ and so requires Python Wheels to simply install (I think RasberryPi wheels might be hard to come by though - and I imagine that might be a common piece of hardware to run camply).

Personally - I think let's just go with numeric IDs - camply users should be familiar with searching for their recreation areas by name in order to retrieve the numeric ID: `camply recreation-areas --provider GoingToCamp --search Tacoma`

Maybe a good numbering system is a prefix for State / Region: i.e. every campground gets a 4 digit code - California is the 5th State in Alphabetical order so it gets campground IDs like 5001 🤷

**Review by juftin — APPROVED** [review]
Nice!

The only thing left to do is squash merge this and give it a GitMoji flavored commit message described here: https://juftin.com/camply/contributing/#releasing

**Comment by acaloiaro** [comment]
@juftin All set.

**Comment by juftin** [comment]
Oh, @acaloiaro I found a bug :(

```
camply campgrounds --provider GoingToCamp --rec-area 5
```

```
ConnectionError: HTTPSConnectionPool(host='hamilton.goingtocamp.com', port=443): Max retries exceeded with url: /api/resourcelocation/rootmaps (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x1297ff6d0>: Failed to establish a new
connection: [Errno 8] nodename nor servname provided, or not known'))
```

**Comment by juftin** [comment]
The following `GoingToCamp` providers are broken for listing campgrounds: 4, 5, 7,

**Comment by acaloiaro** [comment]
@juftin That's a bummer

Hamilton (4) appears to be down: https://hamilton.goingtocamp.com/
Yukon (5) appears to have made an some sort of breaking change to their API server instance and is no longer supplying JSON at the expected URLs
Oroville Parks (7) is no longer listed on GTC's website, so we can assume they're no longer a customer of GTC

My inclination is to delist all of these (they all previously worked).

**Comment by juftin** [comment]
agreed - do we want to re-number them again?

**Pull request merged**